lib/core/src/sdk.rs

```diff
@@ -4031,6 +4050,115 @@ impl LiquidSdk {
+            .sideswap_service
+            .subscribe_price_stream(asset_id, req.send_amount_sat)
+            .await?;
+        Ok(PrepareAssetSwapResponse { asset_swap })
+    }
+
+    pub async fn execute_asset_swap(
+        &self,
+        req: &ExecuteAssetSwapRequest,
+    ) -> Result<ExecuteAssetSwapResponse, PaymentError> {
+        self.sideswap_service.clone().start().await?;
+
+        let req_swap = &req.prepare_response.asset_swap;
+        let Some(ongoing_swap) = self.sideswap_service.get_current_price().await else {
+            return Err(PaymentError::generic(
+                "Cannot execute asset swap: swap not found.",
+            ));
+        };
+
+        ensure_sdk!(
+            ongoing_swap.fees_sat <= req_swap.fees_sat,
+            PaymentError::InvalidOrExpiredFees
+        );
+
+        ensure_sdk!(
+            req_swap.recv_amount == ongoing_swap.recv_amount
+                && req_swap.send_amount_sat == ongoing_swap.send_amount_sat,
+            PaymentError::generic(&format!(
+                "Asset swap amount mismatch: prev - send {} recv {} / new - send {} recv {}",
+                req_swap.send_amount_sat,
+                req_swap.recv_amount,
+                ongoing_swap.send_amount_sat,
+                ongoing_swap.recv_amount
+            ))
+        );
+
+        ensure_sdk!(
+            self.get_info().await?.wallet_info.balance_sat
+                >= ongoing_swap.fees_sat + req_swap.send_amount_sat,
+            PaymentError::InsufficientFunds
+        );
+
+        let ExecuteSwapResponse {
+            tx_id,
+            recv_address: destination,
+        } = self.sideswap_service.execute_swap().await?;
+        self.sideswap_service.stop().await;
+
+        // We insert a pseudo-tx in case LWK fails to pick up the new mempool tx for a while
+        // This makes the tx known to the SDK (get_info, list_payments) instantly
+        let tx_data = PaymentTxData {
+            tx_id: tx_id.clone(),
+            timestamp: Some(utils::now()),
+            amount: req_swap.recv_amount,
+            fees_sat: ongoing_swap.fees_sat,
+            payment_type: PaymentType::Send,
+            is_confirmed: false,
+            unblinding_data: None,
+            asset_id: req_swap.asset_id.clone(),
+        };
+
+        self.persister.insert_or_update_payment(
+            tx_data.clone(),
+            Some(PaymentTxDetails {
+                tx_id: tx_id.clone(),
+                destination: destination.clone(),
+                ..Default::default()
+            }),
+            false,
+        )?;
+        self.emit_payment_updated(Some(tx_id)).await?; // Emit Pending event
+
+        let asset_info = self
+            .persister
+            .get_asset_metadata(&req_swap.asset_id)?
+            .map(|ref am| AssetInfo {
+                name: am.name.clone(),
+                ticker: am.ticker.clone(),
+                amount: am.amount_from_sat(req_swap.send_amount_sat),
```

Comment:
Isn't `send_amount_sat` the L-BTC we're sending? So this should be `req.swap.recv_amount` the asset amount in satoshi?